Index: AsmWriter.cpp
===================================================================
--- AsmWriter.cpp	(revision 270635)
+++ AsmWriter.cpp	(working copy)
@@ -2718,8 +2718,8 @@
     Out << "\n";
     PrintLLVMName(Out, BB->getName(), LabelPrefix);
     Out << ':';
-  } else if (!BB->use_empty()) {      // Don't print block # of no uses...
-    Out << "\n; <label>:";
+  } else {      // Print block # for unnamed basic blocks.
+    Out << "\n";
     int Slot = Machine.getLocalSlot(BB);
     if (Slot != -1)
       Out << Slot << ":";
